engine/src/main/java/com/datatorrent/stram/plan/logical/LogicalPlan.java

```diff
@@ -2424,4 +2425,28 @@ public int hashCode()
     return result;
   }
 
+  /**
+   * Iterate over all registered {@see Visitor} and invoke visit method
+   * for operator and stream. First visitOperator is called for all operators
+   * in the DAG followed by visitStream is called for all the streams in the
+   * DAG.
+   */
+  public void visitDAG()
+  {
+    Collection<Visitor> visitors = getAttributes().get(VISITORS);
+    if (visitors == null) {
+      return;
+    }
+    for (Visitor visitor : visitors) {
+      LOG.info("Using visitor {}", visitor);
+      visitor.preVisitDAG(this);
+      for (OperatorMeta ometa : operators.values()) {
+        visitor.visitOperator(ometa);
+      }
+      for (StreamMeta smeta : streams.values()) {
```

Comment:
So there is no prescribed order in which the streams are visited? If somebody wanted to ensure that starting from the input operators, an operator that is closer to the input in the graph is visited before an operator that is further, they would have to handle that in the visitor implementation itself.